frontend/src/components/program/DialogActivityCreate.vue

```diff
@@ -276,25 +292,44 @@ export default {
         }
       )
     },
+    isActivitySource(value) {
+      if (value?.startsWith(window.location.origin)) {
+        const url = value.substring(window.location.origin.length)
+        const match = router.matcher.match(url)
+        return match.name === 'activity'
+      }
+      return false
+    },
     async getCopyActivitySource(url) {
-      if (url?.startsWith(window.location.origin)) {
+      if (this.isActivitySource(url)) {
         url = url.substring(window.location.origin.length)
         const match = router.matcher.match(url)
+        const scheduleEntry = await this.api
+          .get()
+          .scheduleEntries({ id: match.params['scheduleEntryId'] })
 
-        if (match.name === 'activity') {
-          const scheduleEntry = await this.api
-            .get()
-            .scheduleEntries({ id: match.params['scheduleEntryId'] })
-
-          return await scheduleEntry.activity()
-        }
+        return await scheduleEntry.activity()
       }
       return null
     },
     async clearClipboard() {
       await navigator.clipboard.writeText('')
       this.refreshCopyActivitySource()
     },
+    /// Handles the Paste event which still can be used if the user doesnt grant permission
+    async pasteHandler(event) {
+      const url = (event.clipboardData || window.clipboardData).getData('text')
+      if (!this.isActivitySource(url) || this.copyActivitySourceUrlShowPopover) return
+      if (this.copyActivitySource == null) {
+        const copyActivitySource = await this.getCopyActivitySource(url)
```

Comment:
could you use another name for the local constant and the instance property?

Reply:
done